document-portal/document-portal-fuse.c

```diff
@@ -167,12 +167,15 @@ G_LOCK_DEFINE (domain_inodes);
 
 typedef struct {
   gint ref_count; /* atomic */
+  gint use_count; /* atomic */
```

Comment:
How's the use count any different from the refcount? Can you please clarify why is it necessary to track them separatedly?

Reply:
because it gets kept alive and the fd gets "revived". the _use and _release calls are not matching the _ref and _unref.

Reply:
see https://github.com/flatpak/xdg-desktop-portal/pull/1159/files#diff-2f05bdfc1f7bed1308fa7848357777aad2cadd1d6133d92678296ef2697c4e49R2122

(I have an updated version of that patch with more consistent formatting but the same semantics)